Detectors/EMCAL/reconstruction/src/CaloRawFitter.cxx

```diff
@@ -0,0 +1,288 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file CaloRawFitter.cxx
+/// \author Hadi Hassan (hadi.hassan@cern.ch)
+
+#include "FairLogger.h"
+
+// ROOT sytem
+#include "TMath.h"
+
+#include "EMCALReconstruction/AltroDecoder.h"
+#include "EMCALReconstruction/CaloFitResults.h"
+#include "DataFormatsEMCAL/Constants.h"
+
+#include "EMCALReconstruction/CaloRawFitter.h"
+
+using namespace o2::emcal;
+
+//Default constructor
+CaloRawFitter::CaloRawFitter(const char* name, const char* nameshort) : mMinTimeIndex(-1),
+                                                                        mMaxTimeIndex(-1),
+                                                                        mAmpCut(4),
+                                                                        mNsamplePed(3),
+                                                                        mIsZerosupressed(false),
+                                                                        mName(name),
+                                                                        mNameShort(nameshort),
+                                                                        mAlgo(FitAlgorithm::NONE),
+                                                                        mL1Phase(0),
+                                                                        mAmp(0)
+{
+  for (int i = 0; i < constants::ALTROMAXSAMPLES; i++) {
+    mReversed[i] = 0;
+  }
+}
+
+void CaloRawFitter::setTimeConstraint(const int min, const int max)
+{
+
+  if ((min > max) || min > constants::ALTROMAXSAMPLES || max > constants::ALTROMAXSAMPLES) {
+    LOG(WARN) << "Attempt to set Invalid time bin range (Min , Max) = (" << min << ", " << max << "), Ingored";
+  } else {
+    mMinTimeIndex = min;
+    mMaxTimeIndex = max;
+  }
+}
+
+unsigned short CaloRawFitter::maxAmp(const unsigned short* data, const int length) const
+{
+  unsigned short tmpmax = data[0];
+
+  for (int i = 0; i < length; i++) {
```

Comment:
why not starting from i=1?

Reply:
Should use gsl::span<double> as function argument and std::max_element from <algorithm> for max amplitude search.